ext/sockets/sockets.c

```diff
@@ -1402,7 +1402,8 @@ PHP_FUNCTION(socket_recvfrom)
 
 	/* overflow check */
 	/* Shouldthrow ? */
-	if ((arg3 + 2) < 3) {
+
+	if (arg3 <= 0 || arg3 > ZEND_LONG_MAX - 1) {
```

Comment:
The original code also _intended_ to disallow `ZEND_LONG_MAX-1`, not sure why. I think what you did is fine though.